### DISCUSSION TITLE
Where comes the connection between libary and entity from?

### DISCUSSION THREAD

**Discussion by ToMoHH** [andrew-codechimp/HA-Battery-Notes#537] [General]
As a home assistant beginner, I don't yet understand where the link to the entity comes from. 10 of my 20 battery devices were found.

Where is the type on the entity that is also in the library? 
Unfortunately, I cannot add the missing devices manually because only found entities are available for selection.

I'm looking forward to an explanation.

**Comment by andrew-codechimp** [comment]
The library looks up the manufacturer and model of a device which are displayed in the top left of device info. 

An entity must have a device for battery notes to associate with. If you click on an entity and use the right hand menu you should see device info. If you don't then it's a standalone entity which battery notes doesn't support.

The other possibility is it's not exposing a battery correctly. You can list all devices by adding the following to your configure.yaml

battery_notes:
  show_all_devices: True

Then when you add a device manually it will show a lot more.

**Reply by ToMoHH** [reply]
OK, that's probably where my knowledge ends. 
Apparently the entities cannot be used for my window contact sensors. 
They was imported from a gateway (CCU Homematic).

There ist no extra info for the entitiy.
Can I manually modify entities myself so that they appear in Battery Notes?

![Zwischenablage01](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/2088469/309b35cb-0f62-4902-b217-2c536d410a29)

**Comment by andrew-codechimp** [comment]
Unfortunately if an entity isn't associated with a device there's not much you can do, very odd to have an integration not make a device to group the entities together.
Although technically possible to edit some data files to fake this I wouldn't advise it as thats messing with HA internals and easy to break and no guarantee it would stay that way.

**Reply by fgonza2** [reply]
this is not an 'odd' case.   Many, many home assistant sensors are standalone entities, all the stuff coming from 433Mhz sensors and other integrations come as stand alone entities.   Unfortunately not everything is a well organized device.  The reason i use HA is to be able to integrate many sensors and many are not devices, but a collection o entities.
